examples/token-airdrop/src/example.ts

```diff
@@ -0,0 +1,306 @@
+     * expired.
+     */
+    rpcSubscriptions,
+});
+
+/**
+ * CU LIMIT ESTIMATOR
+ * We create a compute unit limit estimator which will be used by the transaction executor to
+ * estimate the CU limit for each transaction message before sending it.
+ * It does this by simulating the transaction message using the RPC we created earlier.
+ */
+const estimateCULimit = estimateComputeUnitLimitFactory({ rpc });
+// We multiply the simulated limit by 1.1 to add a 10% buffer
+async function estimateWithMultiplier(...args: Parameters<typeof estimateCULimit>): Promise<number> {
+    const estimate = await estimateCULimit(...args);
+    return Math.ceil(estimate * 1.1);
+}
+/**
+ * This helper will take the estimate and use it to update the provisory CU limit instruction
+ * that we included in the planner base transaction message.
+ */
+const estimateAndSetCULimit = estimateAndUpdateProvisoryComputeUnitLimitFactory(estimateWithMultiplier);
+
+/**
+ * TRANSACTION EXECUTOR
+ * The transaction executor is responsible for executing the transaction plan created by the
+ * transaction planner.
+ */
+const transactionExecutor = createTransactionPlanExecutor({
+    /**
+     * This function is called to actually execute each transaction message created by the transaction planner.
+     * It is responsible for signing, sending, and confirming the transaction message.
+     * It only needs to deal with one transaction message.
+     */
+    async executeTransactionMessage(message, config) {
+        const abortSignal = config ? config.abortSignal : undefined;
+
+        /**
+         * We fetch the latest blockhash at this point, so that it is the latest possible
+         * when the transaction is sent.
+         */
+        const { value: latestBlockhash } = await rpc.getLatestBlockhash().send({ abortSignal });
+        const updatedMessage = await pipe(
+            message,
+            tx => setTransactionMessageLifetimeUsingBlockhash(latestBlockhash, tx),
+            /**
+             * We use the helper that we created to estimate and set the CU limit.
+             * This returns an updated transaction message with the CU limit set.
+             * Recall that this updates the provisory CU limit instruction that was added in the planner.
+             * We don't add new instructions in the executor, but we can modify an existing one.
+             */
+            tx => estimateAndSetCULimit(tx, { abortSignal }),
+        );
+
+        // Sign this updated transaction message with any signers included in its instructions
+        const signedTransaction = await signTransactionMessageWithSigners(updatedMessage, { abortSignal });
+
+        const signature = getSignatureFromTransaction(signedTransaction);
+        log.info({ signature }, `[transaction executor] Sending transaction`);
+
+        // Send and confirm the transaction using the helper we created earlier
+        assertIsTransactionWithBlockhashLifetime(signedTransaction);
```

Comment:
This may no longer be needed since we're setting the lifetime in the same function.

Reply:
Unfortunately it is because `signTransactionMessageWithSigners`​ only returns `TransactionWithLifetime`​ to account for modifying signers

I wonder if we could type-juggle it into returning the precise lifetime when there's only non-modifying signers though